linode_api4/objects/tag.py

```diff
@@ -47,22 +59,55 @@ def _api_get(self):
     @property
     def objects(self):
         """
-        Returns a list of objects that have been given this tag.  Right now this
-        is only Linodes, but in the future this may expand to be more.
+        Returns a list of objects with this Tag.  This list may contain any
+        taggable object type.
         """
-        self._get_raw_objects()
+        data = self._get_raw_objects()
+
+        return PaginatedList.make_paginated_list(data, self._client, TaggedObjectProxy,
+                                                 page_url=type(self).api_endpoint.format(**vars(self)))
+
+
+class TaggedObjectProxy:
+    """
+    This class accepts an object form a list of Tagged objects and returns
+    the correct type of object based on the response data.
+
+    .. warning::
+
+       It is incorrect to instantiate this class.  This class is a proxy for the
+       enveloped objects returned from the tagged objects collection, and should
+       only be used in that context.
+    """
+    id_attribute = 'type' # the envelope containing tagged objects has a `type` field
+                          # that defined what type of object is in the envelope.  We'll
+                          # use that as the ID for the proxy class so ``make_instance``
+                          # below can easily tell what type it should actually be
+                          # making and returning.
+
+    @classmethod
+    def make_instance(cls, id, client, parent_id=None, json=None):
+        """
+        Overrides Base's ``make_instance`` to allow dynamic creation of objects
+        based on the defined type in the response json
+
+        :param cls: The class this was called on.
```

Comment:
```suggestion
        :param cls: The class this was called on
```

Either remove this period or add a period to the end of all param statements.